### PR TITLE
feat(orc8r): Periodic_Update_Changes_for_Orc8r

### PR DESCRIPTION
**Pull request by rupa455wavelabs**
## Summary

Added support to fetch the orchestrator changes periodically 

## Test Plan
Added Unit test case for the added function 

test

## Additional Information

Signed-off-by: rupa455wavelabs <rupa.seshaveni@wavelabs.ai>

### CODE REVIEW

File: lte/gateway/python/magma/subscriberdb/tests/client_tests.py [hunk 1]
```diff
@@ -288,6 +300,14 @@ def get_all_subscribers(self):
             ),
         ]
 
+    @ unittest.mock.patch(
```
Comment (ardzoht) [inline lte/gateway/python/magma/subscriberdb/tests/client_tests.py]: this should not be separated by whitespace

File: lte/gateway/python/magma/subscriberdb/client.py [hunk 2]
```diff
@@ -70,6 +86,7 @@ def __init__(
         subscriber_page_size: int,
         sync_interval: int,
         grpc_client_manager: GRPCClientManager,
+
```
Comment (ardzoht) [inline lte/gateway/python/magma/subscriberdb/client.py]: there should not be separation between last param and enclosing parentheses on a function

File: lte/gateway/python/magma/subscriberdb/tests/client_tests.py [hunk 3]
```diff
@@ -288,6 +300,14 @@ def get_all_subscribers(self):
...
+        'magma.common.service_registry.ServiceRegistry.get_rpc_channel',
+    )
+    def list_suci_profiles(self):
+        return [
```
Comment (ardzoht) [inline lte/gateway/python/magma/subscriberdb/tests/client_tests.py]: this should only be returning `SuciProfileList` and not a list of `SuciProfileList` right? what is the purpose of this function? I don't see it being used on the tests

**Review by ardzoht — CHANGES_REQUESTED** [review]
there are some formatting changes needed to be applied, I'd suggest going through the steps on https://github.com/magma/magma/tree/master/lte/gateway/docker/python-precommit to fix those automatically

File: lte/gateway/python/magma/subscriberdb/tests/client_tests.py (hunk 1 above)
Comment (rupa455wavelabs) [reply lte/gateway/python/magma/subscriberdb/tests/client_tests.py]: Hi @ardzoht ,
updated the format as suggested.

File: lte/gateway/python/magma/subscriberdb/tests/client_tests.py (hunk 3 above)
Comment (rupa455wavelabs) [reply lte/gateway/python/magma/subscriberdb/tests/client_tests.py]: yes @ardzoht , I resolved it.

File: lte/gateway/python/magma/subscriberdb/client.py (hunk 2 above)
Comment (rupa455wavelabs) [reply lte/gateway/python/magma/subscriberdb/client.py]: resolved as suggested

File: lte/gateway/python/magma/subscriberdb/client.py [hunk 4]
```diff
@@ -244,6 +264,51 @@ async def _get_all_subscribers(self) -> Optional[CloudSubscribersInfo]:
         )
         return subscribers_info
 
+    async def _list_suciprofiles(self) -> Optional[CloudSuciProfilesInfo]:
+        subscriberdb_cloud_client = self._grpc_client_manager.get_client()
+        suci_profiles = SuciProfile()
+        while not suci_profiles:
+
```
Comment (ardzoht) [inline lte/gateway/python/magma/subscriberdb/client.py]: remove this blank line

**Review by ardzoht — COMMENTED** [review]
left minor comment, please address it and I'll approve

File: lte/gateway/python/magma/subscriberdb/client.py (hunk 4 above)
Comment (rupa455wavelabs) [reply lte/gateway/python/magma/subscriberdb/client.py]: resolved as suggested

**Review by ardzoht — APPROVED** [review]
lgtm, tagging @hcgatewood for agw-orc8r streaming context

**Review by ardzoht — CHANGES_REQUESTED** [review]
Looks like there is merge conflicts, please resolve and address them

**Review by hcgatewood — APPROVED** [review]
Skimmed, seems reasonable to me 👍

**Comment by rupa455wavelabs** [comment]
> Looks like there is merge conflicts, please resolve and address them

Hi @ardzoht, I resolved the merge conflicts.

File: lte/gateway/python/magma/subscriberdb/client.py [hunk 5]
```diff
@@ -69,6 +85,7 @@ def __init__(
         store: SqliteStore,
         subscriber_page_size: int,
         sync_interval: int,
+        suciprofile_db_dict: dict,
```
Comment (ardzoht) [inline lte/gateway/python/magma/subscriberdb/client.py]:
```suggestion
        suciprofile_db_dict: Dict,
```
this should be coming from `from typing import Dict`

**Review by ardzoht — CHANGES_REQUESTED** [review]
one pending change and I'll approve

File: lte/gateway/python/magma/subscriberdb/client.py (hunk 5 above)
Comment (rupa455wavelabs) [reply lte/gateway/python/magma/subscriberdb/client.py]: Updated as suggested.

**Review by ardzoht — APPROVED** [review]
lgtm

**Pull request merged**